src/main/java/com/robinpowered/sdk/model/adapter/IdentifiableAdapter.java

```diff
@@ -0,0 +1,52 @@
+package com.robinpowered.sdk.model.adapter;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.robinpowered.sdk.model.IbeaconIdentifier;
+import com.robinpowered.sdk.model.Identifiable;
+import com.robinpowered.sdk.model.Identifier;
+
+import java.lang.reflect.Type;
+
+/**
+ * Handles adapting {@link com.robinpowered.sdk.model.Identifiable} from JSON representations.
+ */
+public class IdentifiableAdapter implements JsonDeserializer<Identifiable>,
+        JsonSerializer<Identifiable> {
+
+    @Override
+    public Identifiable deserialize(
+            JsonElement json,
+            Type typeOfT,
+            JsonDeserializationContext context
+    ) throws JsonParseException {
+
+        JsonObject object = json.getAsJsonObject();
+        boolean isIBeacon = object.has("device_id")
+                && object.has("major")
+                && object.has("minor")
+                && object.has("uuid");
```

Comment:
If we know that the device is of the identifier type, than we can check the `interface` attribute on the object. If its `"robin-ibeacon"` than you 100% know explicitly that it's that interface type and to deserialize it as such. 

Theoretically its possible for multiple different interfaces to share common fields, so it's not safe to just rely on a few fields being presence, such as `major`.